server.js

```diff
@@ -26,9 +21,145 @@ const app = express()
 app.use(cors())
 app.use(express.json())
 
-// Start defining your routes here
-app.get('/', (req, res) => {
-  res.send('Hello world')
+const Tracks = mongoose.model("Tracks", {
+  index: Number,
+  title: String,
+  artist: String,
+  genre: String,
+  year: Number,
+  bpm: Number,
+  energy: Number,
+  danceability: Number,
+  dB: Number,
+  liveness: Number,
+  valence: Number,
+  duration: Number,
+  acousticness: Number,
+  speechiness: Number,
+  pop: Number,
+})
+
+//resets database by first deleting the previous data with deletemany({}) and then it sets the new data through save() function to avoid duplication of the data everytime the database updates.
+if (process.env.RESET_DB) {
+  const seedDataBase = async () => {
+    await Tracks.deleteMany({})
+
+    spotifyTopTracks.forEach((item) => {
+      const newTrack = new Tracks(item)
+      newTrack.save()
+    })
+  }
+
+  seedDataBase()
+}
+
+// Our own middleware that checks if the database is connected before going forward to our endpoints
+app.use((req, res, next) => {
+  if (mongoose.connection.readyState === 1) {
+    next()
+  } else {
+    res.status(503).json({ error: "Service unavailable" })
+  }
+})
+
+app.get("/", (req, res) => {
+  res.send(
+    "This is my spotify top tracks from the billboard between 2010 - 2019. Type the endpoint /tracks to get the total data to get started!"
+  )
+})
+//See all possible endpoints
+app.get("/endpoints", (req, res) => {
+  res.send(listEndpoints(app))
+})
+
+//Get all 603 song objects from the data array
+app.get("/tracks", async (req, res) => {
+  const tracks = await Tracks.find(req.query)
+  res.json(tracks)
+})
+
+//get a specific song by index between 1-603
+app.get("/tracks/index/:index", async (req, res) => {
```

Comment:
Is there a reason you chose index instead of searching for the song by the id of the song?